server/backend/database/mongo/client.go

```diff
@@ -1149,62 +1149,170 @@ func (c *Client) UpdateAndFindMinSyncedTicket(
+		for _, vvi := range versionVectorInfos {
+			if clientInfo.ID == vvi.ClientID {
+				clientVersionVector = vvi.VersionVector
+
+				continue
+			}
+
+			if minVersionVector == nil {
+				minVersionVector = vvi.VersionVector
+
+				continue
+			}
+
+			minVersionVector = minVersionVector.Min(vvi.VersionVector)
+		}
 	}
 
-	if syncedSeqInfo.ServerSeq == change.InitialServerSeq {
-		return nil, nil
+	// 03. if there's more than one version vector in db.version vector table which is not client's
+	if minVersionVector != nil {
+		actorID, err := clientInfo.ID.ToActorID()
+
+		if err != nil {
+			return nil, err
+		}
+
+		clientLamport := versionVector.VersionOf(actorID)
+		// 03-1. Filter detached client's lamport if exsits
+		filteredVersionVector = versionVector.Filter(minVersionVector.Keys())
+
+		if filteredVersionVector.VersionOf(actorID) <= 0 {
+			filteredVersionVector.Set(actorID, clientLamport)
+		}
+
+		minVersionVector = minVersionVector.Min(filteredVersionVector)
+		// 04. if there's no version vector in db.version vector or there's only one version vector which is current client's
+	} else {
+		// 04-1. if there's no version vector in db.version vector table
+		// which means there's nothing to filter and min version vector is the vv which current client just passed.
+		filteredVersionVector = versionVector
+
+		if clientVersionVector == nil {
+			minVersionVector = versionVector
+		} else {
+			minVersionVector = clientVersionVector
+		}
 	}
 
-	// 03. find the version vector of the min synced seq.
-	// TODO(hackerwins): We need to find the min synced seq of the changes.
-	// min synced seq of the changes is the equivalent of LCA in the dependency graph.
-	result = c.collection(ColChanges).FindOne(ctx, bson.M{
-		"project_id": docRefKey.ProjectID,
-		"doc_id":     docRefKey.DocID,
-		"server_seq": syncedSeqInfo.ServerSeq,
-	})
-	if errors.Is(result.Err(), mongo.ErrNoDocuments) {
-		return nil, nil
+	err = c.UpdateVersionVector(ctx, clientInfo, docRefKey, filteredVersionVector)
+
+	if err != nil {
+		return nil, err
 	}
-	if result.Err() != nil {
-		return nil, fmt.Errorf("find server seq's change: %w", result.Err())
+
+	return minVersionVector, nil
+}
+
+// UpdateVersionVector updates the given version vector of the given client
+func (c *Client) UpdateVersionVector(
+	ctx context.Context,
+	clientInfo *database.ClientInfo,
+	docRefKey types.DocRefKey,
+	versionVector time.VersionVector) error {
+	isAttached, err := clientInfo.IsAttached(docRefKey.DocID)
+	if err != nil {
+		return err
 	}
 
-	changeInfo := database.ChangeInfo{}
-	if err := result.Decode(&changeInfo); err != nil {
-		return nil, fmt.Errorf("decode change: %w", err)
+	if !isAttached {
+		cursor, err := c.collection(ColVersionVector).Find(ctx, bson.M{
+			"project_id": docRefKey.ProjectID,
+			"doc_id":     docRefKey.DocID,
+		})
+		if err != nil {
+			return fmt.Errorf("find all version vectors: %w", err)
+		}
+		defer cursor.Close(ctx)
```

Comment:
Do we need to call this function?

Reply:
I added it just in case we needed it. Is it okay not to have it?